src/gauging.jl

```diff
@@ -59,172 +86,96 @@ function vidal_gauge(
     ψvdst = replaceinds(ψvdst, edge_ind, edge_ind_sim)
     ψvdst = replaceinds(ψvdst * V, commoninds(V, S), new_edge_ind)
 
-    setindex_preserve_graph!(ψ_vidal, ψvsrc, vsrc)
-    setindex_preserve_graph!(ψ_vidal, ψvdst, vdst)
+    setindex_preserve_graph!(updated_ψ, ψvsrc, vsrc)
+    setindex_preserve_graph!(updated_ψ, ψvdst, vdst)
 
     S = replaceinds(
       S,
       [commoninds(S, U)..., commoninds(S, V)...] =>
         [new_edge_ind..., prime(new_edge_ind)...],
     )
-    bond_tensors[e] = S
+    updated_bond_tensors[e] = S
   end
 
-  return ψ_vidal, bond_tensors
+  return VidalITensorNetwork(updated_ψ, updated_bond_tensors)
 end
 
-"""Use an ITensorNetwork ψ in the symmetric gauge and its mts to put ψ into the vidal gauge. Return the bond tensors and ψ_vidal."""
-function vidal_gauge(
-  ψ::ITensorNetwork,
-  pψψ::PartitionedGraph,
-  mts;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  edges=NamedGraphs.edges(ψ),
-  svd_kwargs...,
+function update(
+  ψv::VidalITensorNetwork;
+  (cache!)=nothing,
+  cache_update_kwargs=default_cache_update_kwargs(cache!),
+  kwargs...,
 )
-  bond_tensors = initialize_bond_tensors(ψ)
-  return vidal_gauge(
-    ψ,
-    pψψ,
-    mts,
-    bond_tensors;
-    eigen_message_tensor_cutoff,
-    regularization,
-    edges,
-    svd_kwargs...,
-  )
-end
+  if isnothing(cache!)
+    cache! = Ref(default_norm_cache(itensornetwork(ψv)))
+  end
 
-"""Put an ITensorNetwork into the vidal gauge (by computing the message tensors), return the network and the bond tensors. Will also return the mts that were constructed"""
-function vidal_gauge(
-  ψ::ITensorNetwork;
-  eigen_message_tensor_cutoff=10 * eps(real(scalartype(ψ))),
-  regularization=10 * eps(real(scalartype(ψ))),
-  niters=30,
-  target_canonicalness::Union{Nothing,Float64}=nothing,
-  verbose=false,
-  svd_kwargs...,
-)
-  ψψ = norm_network(ψ)
-  pψψ = PartitionedGraph(ψψ, group(v -> v[1], vertices(ψψ)))
+  cache![] = update(cache![]; cache_update_kwargs...)
 
-  mts = belief_propagation(pψψ; niters, target_precision=target_canonicalness, verbose)
-  return vidal_gauge(
-    ψ, pψψ, mts; eigen_message_tensor_cutoff, regularization, svd_kwargs...
-  )
+  return update(ψv, cache![]; kwargs...)
 end
 
 """Transform from an ITensor in the Vidal Gauge (bond tensors) to the Symmetric Gauge (partitionedgraph, message tensors)"""
-function vidal_to_symmetric_gauge(ψ::ITensorNetwork, bond_tensors::DataGraph)
-  ψsymm = copy(ψ)
-  ψψsymm = norm_network(ψsymm)
-  pψψsymm = PartitionedGraph(ψψsymm, group(v -> v[1], vertices(ψψsymm)))
-  ψsymm_mts = default_bp_cache(pψψsymm)
+function symmetric_gauge(ψv::VidalITensorNetwork)
```

Comment:
It seems like there is a common theme throughout all of these functions that we should settle and that I keep getting confused by, which is: when is a function improving the current gauge/message tensor cache, and then performing a conversion to a different gauge/format, vs. just performing a conversion to a different gauge/format?

My opinion on this is that any function could update the gauge/message tensors or not, and that is chosen with keyword arguments. I think until we settle a general design principle about this, this PR will be very difficult to review and complete.